src/hooks/useResizer.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+
+import { addActions, actions } from "../actions";
+import { defaultState } from "./useTableState";
+
+defaultState.resizedColumns = {};
+
+addActions({
+  resizeColumn: "__resizeColumn__"
+});
+
+const propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      width: PropTypes.number
+    })
+  ),
+  onResizedChange: PropTypes.func
+};
+
+export const useResizer = props => {
+  PropTypes.checkPropTypes(propTypes, props, "property", "useResizer");
+
+  const {
+    columns,
+    hooks: { headers: headerHooks },
+    state: [{ resizedColumns }, setState],
+    getResizerProps,
+    onResizedChange
+  } = props;
+  const currentlyResizingInfo = useRef(undefined);
+
+  useEffect(() => {
+    if (resizedColumns) {
+      Object.keys(resizedColumns).forEach(index => (columns[index].width = resizedColumns[index]));
+    }
+  }, [resizedColumns]);
+
+  const addResizer = (columns, api) => {
+    let bottomLevelColumnCount = 0;
+
+    columns.forEach(column => {
+      if (!column.columns) {
+        const index = bottomLevelColumnCount;
+
+        column.isBottomLevel = true;
+
+        column.resizable = column.resizable !== undefined ? column.resizable : true;
+
+        column.getResizerProps = () => ({
+          ...(column.resizable && {
+            onMouseDown: e => onDragStart(e, column, index),
+
+            // prop overrides
+            ...(getResizerProps && getResizerProps)
```

Comment:
I get what this is meant to do, but I think it would be better just to have the user pass any overrides to the `getResizerProps()` function instead of using a top-level option on the hook. Something like:

```
getResizerProps = ({onMouseDown, ...overrides } = {}) => ({
  onMouseDown: e => {
    onDragStart(e, column, index)
    if (onMouseDown) onMouseDown(e)
  },
  ...overrides
})

Reply:
Yea, i'm cool with that